circleci-artifacts/src/main/java/com/palantir/gradle/utils/circleciartifacts/CircleCiArtifacts.java

```diff
@@ -47,13 +53,40 @@ public final Provider<ArtifactLocation> resolveArtifactLocation(String pathWithi
         Provider<String> circleArtifacts = physicalArtifactPath
                 .map(file -> file.getAsFile().getAbsolutePath())
                 .map(artifacts -> artifacts.replace("/home/circleci/", "/~/"));
+        Provider<String> circleUrl = getVariables()
+                .envVarOrFromTestingProperty("CIRCLE_BUILD_URL")
+                .map(CircleCiArtifacts::extractDomain)
+                .orElse("https://<circle_url>");
+        Provider<String> workflowJobId = getVariables().envVarOrFromTestingProperty("CIRCLE_WORKFLOW_JOB_ID");
 
-        Provider<String> externalLocation = circleProjectUsername
-                .zip(circleProjectReponame, (username, reponame) -> username + "/" + reponame)
-                .zip(circleBuildNum, (fullPath, buildNum) -> fullPath + "/" + buildNum + "/artifacts/")
-                .zip(circleNodeIndex, (fullPath, nodeIndex) -> fullPath + nodeIndex)
+        Provider<String> externalLocation = getZipper()
+                .zip4(
+                        circleProjectUsername,
+                        circleProjectReponame,
+                        circleBuildNum,
+                        circleNodeIndex,
+                        (username, reponame, buildNum, nodeIndex) ->
+                                username + "/" + reponame + "/" + buildNum + "/artifacts/" + nodeIndex)
                 .zip(circleArtifacts, (fullPath, artifactPath) -> fullPath + artifactPath);
 
-        return physicalArtifactPath.zip(externalLocation, ArtifactLocation::of);
+        Provider<String> circleLink = getZipper()
+                .zip4(
+                        circleUrl,
+                        workflowJobId,
+                        circleNodeIndex,
+                        circleArtifacts,
+                        (url, jobId, nodeIndex, artifactPath) ->
+                                url + "/output/job/" + jobId + "/artifacts/" + nodeIndex + artifactPath);
+
+        return getZipper().zip3(physicalArtifactPath, externalLocation, circleLink, ArtifactLocation::of);
+    }
+
+    private static String extractDomain(String url) {
+        try {
+            URL urlObj = new URL(url);
```

Comment:
Make a habit of using `URI` instead of `URL` - `URL` should never be used because it's a footgun that [makes network requests when you use `equals` or `hashCode`](https://errorprone.info/bugpattern/URLEqualsHashCode).